src/coreclr/src/jit/gentree.cpp

```diff
@@ -18308,6 +18315,18 @@ GenTree* Compiler::gtNewMustThrowException(unsigned helper, var_types type, CORI
     return node;
 }
 
+// Returns true for the SIMD Instrinsic instructions that have MemoryLoad semantics, false otherwise
+bool GenTreeSIMD::OperIsMemoryLoad()
```

Comment:
Can you make this method `const` and remove `const_cast` above?

Reply:
Also you added this function inside `#ifdef FEATURE_HW_INTRINSICS` guarded block - but this function should guarded instead by `#ifdef FEATURE_SIMD`

Reply:
> There are existing helper methods such as static int lookupNumArgs(const GenTreeHWIntrinsic* node): https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/hwintrinsic.h#L135 which will return the correct number of arguments for a given node and will attempt to do it as efficiently as possible.

The Value number operation defined by this enum (below), requires that each enum value have a fixed value for arity and commutativity, as we just pass around the value number and then expect to be able to unpack it based upon the VNFunc.

```
enum VNFunc
{
    // Implicitly, elements of genTreeOps here.
    VNF_Boundary = GT_COUNT,
#define ValueNumFuncDef(nm, arity, commute, knownNonNull, sharedStatic) VNF_##nm,
#include "valuenumfuncs.h"
    VNF_COUNT
};

```

Reply:
OK